.github/workflows/ci.yml

```diff
@@ -0,0 +1,47 @@
+name: CI
+on:
+  push:
+    branches: [develop]
+  pull_request:
+
+env:
+  CXXFLAGS: "-fprofile-arcs -ftest-coverage"
+
+jobs:
+  test:
+    name: "Python ${{ matrix.python-version }} / ${{ matrix.os }}"
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        python-version: ["2.7", "3.8"]
+        os: [ubuntu-latest]
+        compiler: [gcc]
+    env:
+      CC: ${{ matrix.compiler }}
+      PY2: ${{ startsWith(matrix.python-version, 2) && 'on' || 'off' }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install dependencies
+        run: |
+          sudo apt-get update -qq
+          sudo apt-get install -qq libboost-all-dev swig libhdf5-serial-dev libeigen3-dev
```

Comment:
You're missing a lot of IMP build deps here -  if you look at the logs you'll see that many modules are not building. On my Ubuntu build images I also install `python-numpy python3-biopython python3-numpy cmake libcgal-dev libcgal-qt5-dev libfftw3-dev libopencv-dev libgsl0-dev python2-dev  python3-dev libann-dev libprotobuf-dev protobuf-compiler python-protobuf python3-protobuf libopenmpi-dev`.

Reply:
I adapted this from Travis. Why don't we need these dependencies over there?

Reply:
Travis just does a very basic build of some of the low-level modules, not all of IMP, just to see if things compile. And it doesn't run any of the tests.